### ISSUE TITLE
Burn 0.18 WGPU: Shader Validation Error & Panic During PPO Loss Calculation

### ISSUE THREAD

**Issue by ManfrediComella** [tracel-ai/burn#3461]
## Description
Upgrading a PPO implementation from Burn 0.17 to 0.18, using WGPU backends (Vulkan/Metal) results in:
1. Shader validation failure due to powf_primitive redefinition
2. Panic during tensor cleanup with "Ordering is bigger than operations"

The same code works in Burn 0.17 with WGPU backend. With the newest version 0.18, it panics.

## How to reproduce
1. Clone the reference repository:
   ```
   git clone https://github.com/yunjhongwu/burn-rl-examples
   cd burn-rl-examples
   ```
2. Modify `main.rs`, make sure to use the `ppo` agent and change the `Backend` from `NdArray` to `Wgpu`:
   ```
   type Backend = Autodiff<Wgpu>;
   type Env = CartPole;

   fn main() {
       let agent = ppo::run::<Env, Backend>(512, false);
       demo_model::<Env>(agent);
   }
   ```
3. Build and run with `cargo`

## Expected Behavior:
Successful training with WGPU backend as in Burn 0.17.

## Error description:
```
{"episode": 0, "reward": 10.0000, "duration": 10}
{"episode": 1, "reward": 22.0000, "duration": 22}

thread 'main' panicked at /home/sieg/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wgpu-25.0.2/src/backend/wgpu_core.rs:1051:30:
wgpu error: Validation Error

Caused by:
  In Device::create_shader_module
    
Shader '' parsing error: redefinition of `powf_primitive`
    ┌─ wgsl:407:4
    │
407 │ fn powf_primitive(lhs: f32, rhs: f32) -> f32 {
    │    ^^^^^^^^^^^^^^ previous definition of `powf_primitive`
    ·
432 │ fn powf_primitive(lhs: f32, rhs: f32) -> f32 {
    │    ^^^^^^^^^^^^^^ redefinition of `powf_primitive`


      redefinition of `powf_primitive`
```

and
```
thread 'main' panicked at library/core/src/panicking.rs:234:5:
panic in a destructor during cleanup
thread caused non-unwinding panic. aborting.
Aborted (core dumped)
```

For the full backtrace click [here!](https://gist.github.com/ManfrediComella/ce4a52c994f84e0c1918dcbc8f37d42c)

**Comment by laggui** [comment]
This should be fixed now

https://github.com/tracel-ai/cubecl/pull/838

**Comment by TitiSkywalker** [comment]
Hi, how can I use the fixed version? I'm using Burn to train a flow matching model, and I get similar problems.

**How to reproduce**
First, you can verify that my code works with Burn 0.17.1:
```bash
git clone https://github.com/TitiSkywalker/Flow-Matching.git
cd Flow-Matching
cargo run
```

Then, you can change `Cargo.toml line 7` to use Burn 0.18.0.
```toml
burn = { version = "0.18.0", features = ["wgpu", "train", "vision"] }
```

Running the program again with Burn 0.18.0, I get 
```
wgpu error: Validation Error

Caused by:
  In Device::create_shader_module
    
Shader '' parsing error: redefinition of `powf_primitive`
    ┌─ wgsl:341:4
    │
341 │ fn powf_primitive(lhs: f32, rhs: f32) -> f32 {
    │    ^^^^^^^^^^^^^^ previous definition of `powf_primitive`
    ·
366 │ fn powf_primitive(lhs: f32, rhs: f32) -> f32 {
    │    ^^^^^^^^^^^^^^ redefinition of `powf_primitive`


      redefinition of `powf_primitive`
```

Which is exactly the above problem. This error happens when I try to backward the losses and update my network:
```Rust
// src/training.rs line 83-87

// Gradients for the current backward pass
let grads = loss.backward();
let grads = GradientsParams::from_grads(grads, &model);
// Update the model using the optimizer
model = optim.step(config.lr, model, grads);
```

Since this bug is fixed two months ago, How can I use the fixed framework from crate? Thank you!

**Comment by laggui** [comment]
@TitiSkywalker the 0.19 release should land this week, but until then you can pull the github revision on main

```toml
burn = { git = "https://github.com/tracel-ai/burn", features = ["wgpu", "train", "vision"] }
```

**Comment by TitiSkywalker** [comment]
Great job! Thank you!
